lib/shopify_api/graphql/response.ex

```diff
@@ -0,0 +1,122 @@
+defmodule ShopifyAPI.GraphQL.Response do
+  @moduledoc """
+  GraphQL Response's Body contains the information such as status_code, 
+  response, metadata[cost, available, restore rate]
+
+  Parses HTTPoison.Response to %ShopifyAPI.GraphQL.Response 
+
+  * `handle/1`
+
+  iex> httpoison = %HTTPoison.Response{
+      body: {:ok,
+          %{
+            "data" => %{
+              "metafield1" => %{
+                "deletedId" => "gid://shopify/Metafield/5256098316335",
+                "userErrors" => []
+                }
+            },
+            "extensions" => %{
+              "cost" => %{
+                "actualQueryCost" => `0,
+                "fields" => [
+                  %{
+                    "definedCost" => 10,
+                    "path" => ["metafield1"],
+                    "requestedChildrenCost" => 0,
+                    "requestedTotalCost" => 10
+                  }
+                ],
+                "requestedQueryCost" => `0,
+                "throttleStatus" => %{
+                  "currentlyAvailable" => 990,
+                  "maximumAvailable" => 1.0e3,
+                  "restoreRate" => 50.0
+                }
+          }
+        }
+  }
+  iex> handle(httpoison)
+  %ShopifyAPI.GraphQL.Response{
+    response: {
+            "data" => %{
+              "metafield1" => %{
+                "deletedId" => "gid://shopify/Metafield/5256098316335",
+                "userErrors" => []
+                }
+            }
+            }, 
+    metadata: %{
+              "cost" => %{
+                "actualQueryCost" => `0,
+                "fields" => [
+                  %{
+                    "definedCost" => 10,
+                    "path" => ["metafield1"],
+                    "requestedChildrenCost" => 0,
+                    "requestedTotalCost" => 10
+                  }
+                ],
+                "requestedQueryCost" => `0,
+                "throttleStatus" => %{
+                  "currentlyAvailable" => 990,
+                  "maximumAvailable" => 1.0e3,
+                  "restoreRate" => 50.0
+                }
+          }, 
+    status_code: 200
+    }
+  }
+
+  """
+
+  alias ShopifyAPI.JSONSerializer
+  alias ShopifyAPI.GraphQL.{JSONParseError, Response}
+
+  @type t :: %ShopifyAPI.GraphQL.Response{
+          response: map(),
+          metadata: map(),
+          status_code: integer(),
+          headers: keyword()
```

Comment:
`headers` will be a regular `list()` - since the symbols in each two-tuple `{key, value}` are strings and not atoms, we can't claim it's a keyword list. Would also make sense to set the default value of the struct to `[]`.